### PR TITLE
Sign transaction by hash

### PR DESCRIPTION
**Pull request by popenta**
If the `version` and `options` field are set accordingly, the transaction will be signed by hash.

The unit test is taken from [this PR](https://github.com/multiversx/mx-sdk-js-core/pull/75).

### CODE REVIEW

File: multiversx_sdk/core/transaction_computer.py [hunk 1]
```diff
@@ -32,10 +35,20 @@ def compute_transaction_fee(self, transaction: ITransaction, network_config: INe
         return int(fee_for_move + processing_fee)
 
     def compute_bytes_for_signing(self, transaction: ITransaction) -> bytes:
+        self._ensure_fields(transaction)
+
+        if transaction.version >= 2 and self.has_options_set_for_hash_signing(transaction):
+            return self.compute_hash_for_signing(transaction)
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transaction_computer.py:41]: We've removed this logic from JS.

https://github.com/multiversx/mx-sdk-js-core/pull/416

Comment (popenta) [reply multiversx_sdk/core/transaction_computer.py]: Indeed. Removed here as well.

File: multiversx_sdk/core/transaction_computer.py [hunk 2]
```diff
@@ -32,10 +35,19 @@ def compute_transaction_fee(self, transaction: ITransaction, network_config: INe
...
         dictionary = self._to_dictionary(transaction)
         serialized = self._dict_to_json(dictionary)
         return serialized
 
+    def compute_hash_for_signing(self, transaction: ITransaction) -> bytes:
+        self._ensure_fields(transaction)
+
+        dictionary = self._to_dictionary(transaction)
+        serialized = self._dict_to_json(dictionary)
+        return keccak.new(digest_bits=256).update(serialized).digest()
```
Comment (bogdan-rosianu) [inline multiversx_sdk/core/transaction_computer.py:49]: the entire function can be:

```
return keccak.new(digest_bits=256).update(self.compute_bytes_for_signing(transaction)).digest()
```

Comment (popenta) [reply multiversx_sdk/core/transaction_computer.py:49]: Indeed. Changed

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**